packages/notifications-flyout/src/behaviors/NotificationFlyoutBehavior.js

```diff
@@ -0,0 +1,163 @@
+/* eslint-disable react/no-unused-prop-types */
+
+import { Component } from "react";
+import PropTypes from "prop-types";
+import { polyfill } from "react-lifecycles-compat";
+
+import parseNotifications from "./parseNotifications";
+
+/** @typedef {import("./parseNotifications").Input} RawNotifications */
+/** @typedef {import("./parseNotifications").ParsedNotification} ParsedNotification */
+/** @typedef {import("./NotificationBehavior").Payload} NotificationBehaviorPayload */
+
+/**
+ * @typedef {Object} Payload
+ * @property {Function} dismissNotification
+ * @property {ParsedNotification[]} notifications
+ * @property {boolean} showUnreadCount
+ * @property {number} unreadCount
+ */
+
+/**
+ * @typedef {Object} Props
+ * @property {function(Payload): import("react").ReactElement} [children]
+ * @property {import("./parseNotifications").Input} [notifications]
+ * @property {number} [unreadCount]
+ */
+
+/**
+ * @typedef {Object} State
+ * @property {string[]} dismissedNotifications An array of notification IDs that have been dismissed
+ * @property {ParsedNotification[]} notifications
+ * @property {string[]} readNotifications An array of notification IDs that have been read
+ */
+
+class NotificationFlyoutBehavior extends Component {
+  static propTypes = {
+    /** Render prop */
+    children: PropTypes.func.isRequired,
+    /** Rendered notifications */
+    notifications: PropTypes.oneOfType([
```

Comment:
I'm unclear on the relationship between `children` and `notifications` here.

Reply:
In short `children` provides a JSX interface for rendering components, while `notifications` provides an array-based interface for rendering components.

Both of the following examples will render the same thing:

```jsx
<NotificationsFlyout>
  <Notification id="1" />
</NotificationsFlyout>
```
```jsx
<NotificationsFlyout notifications={[{ id: "1" }]} />
```